sqlglot/helper.py

```diff
@@ -24,7 +27,17 @@ def list_get(arr, index):
 def ensure_list(value):
     if value is None:
         return []
-    return value if isinstance(value, (list, tuple, set)) else [value]
+    return [*value] if isinstance(value, (list, tuple, set)) else [value]
```

Comment:
I think this introduces loads of copy operations. throughout sqlglot. 

Based on how we use `ensure_list`, I think that's _safe_, but I wonder how it effects performance?

Reply:
agree, we shoudln't do this as it will hurt perf

Reply:
Without this change though we aren't really ensuring it is a list which is confusing.

Reply:
it’s just a bad name. it’s ensure enumerable, the better way is to do list(x) where you actually need a list

Reply:
Ok I reverted this change